src/lib.rs

```diff
@@ -53,6 +54,30 @@ pub fn candidate_cert_dirs() -> impl Iterator<Item = &'static Path> {
     .filter(|p| p.exists())
 }
 
+#[cfg(target_os = "freebsd")]
+pub fn candidate_cert_dirs() -> impl Iterator<Item = &'static Path> {
+    extern "C" {
```

Comment:
I do not know Rust, but can't this function be merged into the above one somehow?

Reply:
@djc Sorry for playing stupid, not a Rust hacker. Is this what you are looking for:
```rust
pub fn candidate_cert_dirs() -> impl Iterator<Item = &'static Path> {
    #[cfg(target_os = "freebsd")]
    let dirs = &[
        "/etc/ssl",
        "/usr/local/etc/ssl",
        "/usr/local/openssl",
    ];

    #[cfg(not(target_os = "freebsd"))]
    let dirs = &[
        "/etc/ssl",
        "/etc/pki/tls/certs",
        "/usr/share/ssl",
        "/usr/local/share/certs",
        "/etc/openssl",
        "/var/ssl",
        "/usr/lib/ssl",
        "/usr/local/openssl",
        "/usr/local/etc/ssl",
    ];

    dirs.iter().map(|s| Path::new(s))
}
```

including the current `#cfg`...